src/Utilities/hooks/useFetchBatched.js

```diff
@@ -1,43 +1,10 @@
+import { useCallback, useMemo } from 'react';
 import usePromiseQueue from './usePromiseQueue';
 
 const useFetchBatched = () => {
   const { isResolving: isLoading, resolve } = usePromiseQueue();
-
-  return {
-    isLoading,
-    fetchBatched: (fetchFunction, total, filter, batchSize = 50) => {
-      const pages = Math.ceil(total / batchSize) || 1;
-
-      const results = resolve(
-        [...new Array(pages)].map(
-          (_, pageIdx) => () =>
-            fetchFunction(filter, { page: pageIdx + 1, per_page: batchSize })
-        )
-      );
-
-      return results;
-    },
-    fetchBatchedInline: (fetchFunction, list, batchSize = 20) => {
-      const pages = Math.ceil(list.length / batchSize) || 1;
-
-      const results = resolve(
-        [...new Array(pages)].map(
-          (_, pageIdx) => () =>
-            fetchFunction(
-              list.slice(batchSize * pageIdx, batchSize * (pageIdx + 1))
-            )
-        )
-      );
-
-      return results;
-    },
-    pageOffsetfetchBatched: (
-      fetchFunction,
-      total,
-      filter,
-      batchSize = 50,
-      pageOffset = 0
-    ) => {
+  const pageOffsetfetchBatched = useCallback(
```

Comment:
Since it's got only one `resolve` dependency in this useCallback, can't we just leave the function in the return `useMemo` object as we for fetchBatched and fetchBatchedInline? Does it necessarily need to be wrapped in separate useCallback?

Reply:
> Does it necessarily need to be wrapped in separate useCallback?

Yes, if it is not it will cause the function to always "change" and the `useEffect` is always executed.